README.md

```diff
@@ -1,3 +1,124 @@
 # cloud-sql-go-connector
 *Warning*: This project is experimental, and is not an officially supported 
 Google product.
+
+The _Cloud SQL Go Connector_ provides IAM authorization to an app's connections 
+to a Cloud SQL instance. It provides connection level authorization only; it 
+_does not_ provide a network path to an instance that doesn't already exist 
+(ie. you will still be unable to connect via Private IP without access to the 
+correct VPC). For more information, see 
+[About the Cloud SQL Auth proxy][about-proxy].
+
+[about-proxy]: https://cloud.google.com/sql/docs/mysql/sql-proxy
+
+The _Cloud SQL Go Connector_ is an experimental new version of the 
+[Cloud SQL proxy dialer](). It's API is considered unstable and may change in 
+the future. Please use at your own risk. 
+
+[proxy-dialer]: https://github.com/GoogleCloudPlatform/cloudsql-proxy/tree/main/proxy#cloud-sql-proxy-dialer-for-go
+
+## Installation
+
+First, clone this repo into a folder relative to your project:
+```sh
+git clone https://github.com/kurtisvg/cloud-sql-go-connector.git
+```
+
+Next, use a _replace_ directive in your `go.mod` that points to the cloned folder:
+```
+replace cloud.google.com/cloudsqlconn => ../cloud-sql-go-connector
+```
+
+## Usage 
+
+This package provides several functions for authorizing and encrypting 
+connections on your behalf. Typically, these functions will pass these 
+connections to the driver while interacting with your database.
+
+The instance connection name for your Cloud SQL instance is always in the 
+format "project:region:instance".
+
+### Credentials 
+
+This repo uses the [Application Default Credentials (ADC)][adc] strategy for 
+providing credentials. Please see the [golang.org/x/oauth2/google][google-auth] 
+documentation for more information in how these credentials are sourced. 
+
+// TODO: Mention customizing source
+
+[adc]: https://cloud.google.com/docs/authentication
+[google-auth]: https://pkg.go.dev/golang.org/x/oauth2/google#hdr-Credentials
+
+### Using the default Dialer
+
+If you don't need to customize your Dialer's behavior, it's convenient to use 
+the package's "Dial" option, which initialized a default dialer for you.  
+<details>
```

Comment:
Making this a subsection in the README seems fine to me, rather than an expandable section.